### ISSUE TITLE
Using an enum variant in the size of a const array shows an error in CLion, but cargo builds without errors

### ISSUE THREAD

**Issue by InfinityGlitch** [intellij-rust/intellij-rust#10463]
## Environment

* **IntelliJ Rust plugin version:** 0.4.194.5382-231
* **Rust toolchain version:** 1.69.0
* **IDE name and version:** CLion 2023.1.2
* **Operating system:** Fedora Linux 38

## Problem description
The problem occurs on this piece of code, which produces no errors when using `cargo build`/`check`, but does show an error in CLion.
```rust
enum Foo {
    Bar
}

const ARRAY: [u8; Foo::Bar as usize] = [];
```
CLion shows an error on the `Foo::Bar` expression, which reads: `Calls in constants are limited to constant functions, tuple structs and tuple variants [E0015] `.
